src/AzureAuth.Test/CommandMainTest.cs

```diff
@@ -384,6 +388,121 @@ public void TestPromptHintPrefixWithoutOption()
+            subject.Client = "e19f71ed-3b14-448d-9346-9eff9753646b";
+            subject.Tenant = "9f6227ee-3d14-473e-8bed-1281171ef8c9";
+
+            subject.EvaluateOptions().Should().BeTrue();
+            subject.CacheFilePath.Should().Be(cacheFilePath);
+        }
+
+        /// <summary>
+        /// The test to evaluate the cache file name when both enviroment variable and option exist.
+        /// Ideally use option first.
+        /// </summary>
+        [Test]
+        [Platform("Win")] // Only valid on Windows
+        public void TestCacheFileOptionWithFilenameFromEnvAndOption()
+        {
+            string filenameFromEnv = "C:\\test\\absolute_from_env.cache";
+            this.envMock.Setup(env => env.Get("AZUREAUTH_CACHE_FILE")).Returns(filenameFromEnv);
+
+            CommandMain subject = this.serviceProvider.GetService<CommandMain>();
+            subject.CacheFilePath = "C:\\test\\absolute_from_option.cache";
+
+            subject.Resource = "f0e8d801-3a50-48fd-b2da-6476d6e832a2";
+            subject.Client = "e19f71ed-3b14-448d-9346-9eff9753646b";
+            subject.Tenant = "9f6227ee-3d14-473e-8bed-1281171ef8c9";
+
+            subject.EvaluateOptions().Should().BeTrue();
+            subject.CacheFilePath.Should().Be("C:\\test\\absolute_from_option.cache");
+        }
+
+        /// <summary>
+        /// The test to evaluate a default cache file name.
+        /// </summary>
+        [Test]
+        [Platform("Win")] // Only valid on Windows
+        public void TestCacheFileOptionWithNoParameter()
+        {
+            CommandMain subject = this.serviceProvider.GetService<CommandMain>();
+            subject.Resource = "f0e8d801-3a50-48fd-b2da-6476d6e832a2";
+            subject.Client = "e19f71ed-3b14-448d-9346-9eff9753646b";
+            subject.Tenant = "9f6227ee-3d14-473e-8bed-1281171ef8c9";
+
+            subject.EvaluateOptions().Should().BeTrue();
+
+            string appData = Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData);
+            string absolutePath = Path.Combine(appData, ".IdentityService", $"msal_{subject.Tenant}.cache");
+            string expected = absolutePath;
+
+            subject.CacheFilePath.Should().Be(expected);
+        }
+
+        /// <summary>
+        /// The test to evaluate a relative cache path,
+        /// which should return false since we only expect an absolute path.
+        /// </summary>
+        [Test]
+        [Platform("Win")] // Only valid on Windows
+        public void TestCacheFileOptionWithRelativePath()
+        {
+            CommandMain subject = this.serviceProvider.GetService<CommandMain>();
+            subject.Resource = "f0e8d801-3a50-48fd-b2da-6476d6e832a2";
+            subject.Client = "e19f71ed-3b14-448d-9346-9eff9753646b";
+            subject.Tenant = "9f6227ee-3d14-473e-8bed-1281171ef8c9";
+
+            string path = "../test/relative.cache";

```

Comment:
This says it's a Windows only test, but it uses non-Windows path separators. Is that correct? :thinking:

Reply:
That's right. I changed it to Windows-style.